tools/clang/unittests/HLSLExec/LongVectors.cpp

```diff
@@ -628,6 +628,21 @@ struct StrictValidation {
 #define DEFAULT_OP_2(OP, IMPL) OP_2(OP, DefaultValidation<T>, IMPL)
 #define DEFAULT_OP_3(OP, IMPL) OP_3(OP, DefaultValidation<T>, IMPL)
 
+#define OP_WITH_OUT_PARAM_1(OP, TYPE, IMPL)                                    \
```

Comment:
Putting this much code inside a macro definition makes me uneasy - especially once we start putting blocks of code in the IMPL parameter.

I think it might be worth spending some time trying to find ways to have this be written more in C++ than in the preprocessor.